src/realm/object-store/sync/sync_session.cpp

```diff
@@ -235,7 +235,7 @@ SyncSession::handle_refresh(const std::shared_ptr<SyncSession>& session)
     };
 }
 
-SyncSession::SyncSession(SyncClient& client, std::shared_ptr<DB> db, SyncConfig config, SyncManager* sync_manager)
+SyncSession::SyncSession(SyncClient& client, std::shared_ptr<DB> db, RealmConfig config, SyncManager* sync_manager)
```

Comment:
This needs to set `scheduler` and `audit_config` to `nullptr` on `m_config`. Trying to use the explicitly set `scheduler` will break since we're not obtaining the Realm instance while inside that scheduler. Audit stuff might not _break_ but we don't really want to be initializing that while performing a client reset regardless.